pages/[id]/edit.js

```diff
@@ -1,37 +1,46 @@
 import { useRouter } from "next/router";
 import useSWR from "swr";
 import SnippetForm from "@/components/SnippetForm";
+import LinkLayout from "@/components/LinkLayout";
 import styled from "styled-components";
+
 export default function EditPage({ defaultTags }) {
   const router = useRouter();
   const { isReady } = router;
   const { id } = router.query;
   const { data: snippet, isLoading, error } = useSWR(`/api/snippets/${id}`);
 
   async function editSnippet(event, snippetData) {
-    try {
-      const { id } = router.query;
-      const response = await fetch(`/api/snippets/${id}`, {
-        method: "PUT",
-        headers: {
-          "Content-Type": "application/json",
-        },
-        body: JSON.stringify(snippetData),
-      });
+    if (snippetData.name !== "" && snippetData.code !== "") {
+      try {
+        const { id } = router.query;
+        const response = await fetch(`/api/snippets/${id}`, {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(snippetData),
+        });
 
-      if (response.ok) {
-        router.push(`/${id}`);
+        if (response.ok) {
+          router.push(`/${id}`);
+        }
+      } catch (error) {
+        console.error("An error occurred:", error);
+        response.status(500).json({ error: "Something went wrong" });
```

Comment:
this seems to be doing nothing, you would expect something like this on a backend route